### ISSUE TITLE
integration til Elasticsearch

### ISSUE THREAD

**Issue by JosefAssad** [JosefAssad/SeMaWi#41]

**Comment by ballerupgis** [comment]
Hurtig, søgbar og skalerbar ledelsesinfo

**Comment by baffioso** [comment]
Vi kunne evt se på mulighederne i denne extension:
[https://m.mediawiki.org/wiki/Extension:CirrusSearch]

**Comment by ClockEnd** [comment]
Spændende
Den 01/12/2015 6.29 AM skrev "baffioso" <notifications@github.com>:

**Comment by JosefAssad** [comment]
Ja, det er den rigtige udvidelse, det er den WikiMedia foundation bruger. En mindre bekymring er, at der sidst blev udgivet ny version i oktober 2014, men det kunne også godt betyde at den bare er stabil.

Denne issue kommer nok til at blive prioriteret højt i 2016 roadmap.

**Comment by ClockEnd** [comment]
Det kræver at vi tager SeMaWi ind på egen server, hvor vi har Elasticsearch kørende? Så er det processen omkring portering af løsningen vi skal have lagt en plan for

**Comment by JosefAssad** [comment]
Ja jeg fornemmer at det vil betyde at SeMaWi skal flyttes. Jeg tror ikke det vil medføre kodemæssige ændringer, så i princippet ender vi nok med at lukke denne issue uden henvisning til et git commit, men vi holder den åben da fx. dokumentatioenn vil skulle opdateres.

**Comment by baffioso** [comment]
vedr. https://github.com/JosefAssad/SeMaWi/issues/94#issuecomment-171630496
Vi er ved at få fyret op i en ny intern fælles GIS-server, som vi deler med Egedal og Furesø kommene. Spørgsmålet er om Wikien skal ligge der eller om vi skal lave en dedikeret Wiki server. Hvad er din holdning @JosefAssad?

**Comment by JosefAssad** [comment]
Root access, en "sane" Linux version, 64 bit (til når vi skifter til docker ref #91), og er der plads til den (hvor meget fylder GIS applikationerne ift. serverens specs)?

Jeg vil være overrasket hvis der ikke er plads til wiki'en på en GIS appserver, og det er nice at bruge det vi har. :) Den er jo ikke specielt ressourcekrævende.

Hvad med data? Kan der være noget i den data wiki'en indeholder som ikke må kune tilgås af andre? Fx. hvis #93 ender med at være noget i vil aktivere.

**Comment by baffioso** [comment]
Går også stærkt ud fra der er kapacitet til wikien på serveren som får Ubuntu 14 #.
Der vil kontraktoplysninger, som ikke må kunne tilgås af de andre kommuner som har adgang til wikien, men der er jo login.

**Comment by JosefAssad** [comment]
Så regner jeg med vi kan starte med test som minimum. Successchancerne er nok på 90% :)

Den server kar adgang til Elasticsearch, yes?

**Comment by baffioso** [comment]
Yes. Kommer med GC2 stacken

**Comment by JosefAssad** [comment]
En praktisk detalje. MW kræver i øjeblikket (via Extension:cirrussearch) Elasticsearch 1.7. Hvad bruger GC2? Regner med at det er en nyere 2.*?

Her har vi to optioner. Enten kører vi separat ES 1.7 instans i denne milestone eller så kan vi vente til MW opdaterer udvidelsen (https://phabricator.wikimedia.org/T122698), som skulle være et højt prioritet men nok ikke vil nås i Q1.

Tanker?

**Comment by baffioso** [comment]
GC2 kører med seneste udgivelse af ES. 

Faldt over denne https://www.elastic.co/blog/loading-wikipedia.

**Comment by JosefAssad** [comment]
Yeah eksemplerne i den blog post anvender API kald som er introduceret af CirrusSearch udvidelsen, jeg forestiller mig at bloggeren bruger en 1.7 ES der.

Jeg begynder at synes at vi skal udskyde denne issue til milestone 2016_02. CirrusSearch kræver ES 1.7, men Kibana kræver ES 2.2. Det betyder at selv hvis vi får ES 1.7 op og køre med denne milestone så kommer den ikke i Kibana, og så har vi to parallele ElasticSearch servers; en til GC2 og en til SeMaWi.

WikiMedia foundation har [opgraderingsopgaven i deres issue tracker](https://phabricator.wikimedia.org/T122698). Jeg har snakket med et par gutter fra deres Discovery Team og de regner med at den bliver implementeret i Q2.

Med mindre i har en use case på at en ES til SeMaWi der ikke snakker med GC2s ES har værdi så vil jeg anbefale at vi venter på den opgradering.

Tanker?

**Comment by baffioso** [comment]
Tænker at vi udskyder til de har ES 2 integration

**Comment by JosefAssad** [comment]
Ny Phabricator sag vi fælger: https://phabricator.wikimedia.org/T133120

**Comment by JosefAssad** [comment]
Mediawiki er opgraderet til at bruge Elasticsearch 2.* men kun kå wmf/1.28 branches.

Efter diskussion med MW udviklere er der åbnet en [Phabricator sag](https://phabricator.wikimedia.org/T146636) for at se om vi kan få en backport til 1.27 som vi gerne vil blive på indtil næste LTS.

Jeg følger sagen.

**Comment by JosefAssad** [comment]
Phabricator sagen er godkendt og Elasticsearch backportes til MW 1.27, men bliver implementeret formentlig i 2017 Q2.
